eng/common/TestResources/New-TestResources.ps1

```diff
@@ -313,25 +312,63 @@ foreach ($templateFile in $templateFiles) {
         }
     }
 
-    foreach ($key in $deploymentOutputs.Keys)
+    if ($OutFile)
     {
-        $value = $deploymentOutputs[$key]
-        
-        if ($CI) {
-            # Treat all ARM template output variables as secrets since "SecureString" variables do not set values.
-            # In order to mask secrets but set environment variables for any given ARM template, we set variables twice as shown below.
-            Write-Host "Setting variable '$key': ***"
-            Write-Host "##vso[task.setvariable variable=_$key;issecret=true;]$($value)"
-            Write-Host "##vso[task.setvariable variable=$key;]$($value)"
-        } else {
-            Write-Host ($shellExportFormat -f $key, $value)
+        if (!$IsWindows)
+        {
+            Write-Host "File option is supported only on Windows"
         }
+
+        # Compute a hash of repository root to scope the test environment file
+        $repositoryRootHash = New-Object System.Text.StringBuilder
```

Comment:
Why has the root? Wouldn't it be easier to just put it somewhere under the root but added to .gitignore?

Reply:
Last time we chatted about it there were some concerns about having files with secrets in the source tree. I'm fine with any option.

Reply:
If they are encrypted I have less concerns with them being in the source tree and kind of prefer it because it makes using them and cleaning them easier.  We should come up with a file extension that we can add to the gitignore files in our repos.

Reply:
Made the change, `test-resources.json.env` is the file name.